### PR TITLE
feat(`evm`): migrate to `alloy-genesis`, rm `ethers-core`

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Migrates to using alloy genesis instead of ethers core types for genesis utils.

## Solution

Removes ethers core, migrates to alloy genesis. 

The `loadAllocs` cheatcode was modified to try reading from a json file which only contains allocs, if the allocs read from the genesis file is empty. This is because now `Genesis` can be deserialized as default, and will not fail if the `alloc` field is missing.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
I noticed we also have this duplicated here:

https://github.com/foundry-rs/foundry/blob/c6790a02bb80b163469e2e3bb0b16ae95ad7334f/crates/anvil/src/genesis.rs

which we can now also replace

**Comment by mattsse** [comment]
failing tests look related

**Comment by Evalir** [comment]
Yeah tests are failing because it seems storage is not being set properly—need to take a look

**Comment by mattsse** [comment]
bump here

File: crates/cheatcodes/src/evm.rs [hunk 1]
```diff
@@ -77,12 +77,18 @@ impl Cheatcode for loadAllocsCall {
...
+        let allocs: HashMap<Address, GenesisAccount> = match genesis {
+            Ok(genesis) => genesis.alloc,
             Err(_) => read_json_file(path)?,
         };
 
+        // If allocs is empty, let's assume we're reading a genesis_accounts.json file.
+        let allocs = match allocs.is_empty() {
+            true => read_json_file::<HashMap<Address, GenesisAccount>>(path)?,
+            false => allocs,
+        };
```
Comment (mattsse) [inline crates/cheatcodes/src/evm.rs:90]: this is redundant no?

I think this should be flipped, try Hashmap first then genesis

**Review by mattsse — CHANGES_REQUESTED** [review]
what's the issue exactly that prompted the other PR

still not clear to me

**Comment by Evalir** [comment]
@mattsse for the issue, see: https://github.com/alloy-rs/alloy/pull/151#issuecomment-1908605365

**Comment by mattsse** [comment]
thanks, no this makes sense to me 🙏

File: crates/cheatcodes/src/evm.rs (hunk 1 above)
Comment (Evalir) [reply crates/cheatcodes/src/evm.rs]: Yep, good call. Flipped the check here: https://github.com/foundry-rs/foundry/pull/6880/commits/598a1d5fc54649ea180fb913980801441960d5a8

File: Cargo.lock [hunk 2]
```diff
@@ -7979,9 +7995,9 @@ checksum = "accd4ea62f7bb7a82fe23066fb0957d48ef677f6eeb8215f372f52e48bb32426"
 
 [[package]]
 name = "vergen"
-version = "8.3.0"
+version = "8.3.1"
```
Comment (DaniPopes) [inline Cargo.lock:7998]: pls try to leave `cargo update`s to the weekly PR for reviewability

File: crates/cheatcodes/src/evm.rs [hunk 3]
```diff
@@ -121,18 +121,12 @@ impl Cheatcode for dumpStateCall {
                     key,
                     GenesisAccount {
                         nonce: Some(val.info.nonce),
-                        balance: val.info.balance.to_ethers(),
-                        code: Some(
-                            val.info.code.clone().unwrap_or_default().original_bytes().to_ethers(),
-                        ),
+                        balance: val.info.balance,
+                        code: Some(val.info.code.clone().unwrap_or_default().original_bytes()),
```
Comment (DaniPopes) [inline crates/cheatcodes/src/evm.rs:125]: as_ref map(original_bytes) unwrap_or_default

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**